### DISCUSSION TITLE
Webview best practice.

### DISCUSSION THREAD

**Discussion by ShaswatChampatiray** [microsoft/vscode-discussions#503] [Extension Development QnA] [answered]
As mentioned in the [Webview API Docs](https://code.visualstudio.com/api/extension-guides/webview#should-i-use-a-webview) we should be sparingly using the webviews as it is resource heavy.
Is there any best practices or benchmark for the number of webviews to be registered in an extension for optimum performance.

**Comment by cmalin-bmc** [comment]
Piggybacking on this question... What is behind the advice to sparingly use webviews? What is the potential impact to the user experience if webviews are used?

**Reply by digitarald** [reply]
Mostly to keep a consistent interaction pattern in VS Code and the fact that native UI components come with lifecycle, a11y, performance, etc considerations built-in.

**Comment by digitarald** [comment, marked as answer]
Afaik there is no specific limit. Best to have the user in mind, as their VS Code keeps growing with rich webviews. Knowing how quick a webview loads and how much data it holds during development would be advisable.

If your extension really depends on the web use to work, it should used them; making sure they are accessible, don't leak memory, adapt to the current theme, etc. What we want to avoid is a proliferation of complex UIs don't fit into VS Code look & feel; but bias towards use of quick picks or other built-in UI elements.

**Reply by cmalin-bmc** [reply]
Thanks for the reply, @digitarald. That's helpful.
